frontend/src/pages/MemberOnboarding.jsx

```diff
@@ -0,0 +1,315 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+import api from '../api';
+import { useNavigate } from 'react-router-dom';
+import { ONBOARDED } from '../constants';
+
+const confirmDiscordEndpoint = (username) => `http://localhost:8000/api/members/confirm-discord/${username}`;
+
+const Container = styled.div`
+  max-width: 500px;
+  margin: 0 auto;
+  padding: 20px;
+  font-family: Arial, sans-serif;
+`;
+
+const Title = styled.h1`
+  text-align: center;
+  color: #333;
+`;
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+`;
+
+const Input = styled.input`
+  margin: 10px 0;
+  padding: 10px;
+  border: 1px solid #ddd;
+  border-radius: 4px;
+`;
+
+const Button = styled.button`
+  margin: 10px 0;
+  padding: 10px;
+  background-color: #0066cc;
+  color: white;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+  &:hover {
+    background-color: #0052a3;
+  }
+`;
+
+const ErrorMessage = styled.p`
+  color: red;
+`;
+
+const DiscordConfirmation = ({ onNext, discordInfo }) => {
+  const [discordUsername, setDiscordUsername] = useState(discordInfo.discordUsername);
+  const [discordId, setDiscordId] = useState(discordInfo.discordId);
+  const [error, setError] = useState('');
+
+  const handleConfirm = async () => {
+    try {
+      const response = await api.get(confirmDiscordEndpoint(discordUsername));
+      setError('Check your Discord for the confirmation code.');
+    } catch (err) {
+      setError('Failed to send confirmation. Please try again.');
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (discordId) {
+      onNext({discordUsername, discordId});
+    } else {
+      setError('Please enter the Discord ID sent to you.');
```

Comment:
I think we should use some randomly generated auth token instead of the discord ID, discord ID's are viewable by anyone the bot will just record the ID though into the DB so we know who is who since user names change.

Reply:
so long as we can also get their discord id (maybe through api, or ideally if bot returns it to us) then yes